code/game/turfs/simulated/walls.dm

```diff
@@ -209,14 +213,34 @@
 			projectile_reflection(Proj)
 			return PROJECTILE_CONTINUE // complete projectile permutation
 
-	//cut some projectile damage here and not in projectile.dm, becouse we need not to all things what are using get_str_dam() becomes thin and weak.
+	//cut some projectile damage here and not in projectile.dm, because we need not to all things what are using get_str_dam() becomes thin and weak.
 	//in general, bullets have 35-95 damage, and they are plased in ~30 bullets magazines, so 50*30 = 150, but plasteel walls have only 400 hp =|
 	//but you may also increase materials thickness or etc.
 	proj_damage = round(Proj.get_structure_damage() / 3)//Yo may replace 3 to 5-6 to make walls fucking stronk as a Poland
 
 	//cap the amount of damage, so that things like emitters can't destroy walls in one hit.
 	var/damage = min(proj_damage, 100)
 
+	if(bulletholes >= 30)
```

Comment:
Why clear holes if they're over 30?
Wouldn't it make more sense only to clear the single oldest one, or to not add new ones, or even just to let them keep building up infinitely ?

Reply:
Well it makes a new object for every hole so I figured an infinite amount of objects would probably lag the game. It's a pretty arbitrary number but I have never been able to even see it get to thirty without the wall breaking.